### PR TITLE
Set COMPILE_ONLY flag on relevant dependencies that appear on DEPLOYMENT_CP and RUNTIME_CP

### PR DESCRIPTION
**Pull request by aloubyansky**

### CODE REVIEW

File: independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/ApplicationDependencyTreeResolver.java [hunk 1]
```diff
@@ -281,15 +281,19 @@ private void collectCompileOnly(CollectRequest collectRtDepsRequest, DependencyN
         int flags = DependencyFlags.DIRECT | DependencyFlags.COMPILE_ONLY;
         while (children != null) {
             for (DependencyNode node : children) {
-                if (appBuilder.getDependency(getKey(node.getArtifact())) == null) {
-                    var dep = newDependencyBuilder(node, resolver).setFlags(flags);
-                    if (getExtensionInfoOrNull(node.getArtifact(), node.getRepositories()) != null) {
+                var extInfo = getExtensionInfoOrNull(node.getArtifact(), node.getRepositories());
+                var dep = appBuilder.getDependency(getKey(node.getArtifact()));
```
Comment (gsmet) [inline independent-projects/bootstrap/maven-resolver/src/main/java/io/quarkus/bootstrap/resolver/maven/ApplicationDependencyTreeResolver.java:285]: FWIW, this `var` thing makes it a lot harder to review PR as you don't have the IDE context. So you can't easily figure out which class it is and have a look if needed.

**Review by ia3andy — APPROVED** [review]
Thanks @aloubyansky it works like a charm!

Tested with Web-Bundler

**Comment by ia3andy** [comment]
@gsmet this is "needed" for the `3.7.3`

**Comment by gsmet** [comment]
I'll try to get it in before I create the backport PR but it's already late and CI is still not finished.

**Pull request merged**